### DISCUSSION TITLE
Auto-purging revisions

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Revisions#920] [Feature Requests]
It might be worth having an option to auto-purge pending revisions after a period of time (3 months, 6 months etc)

Related: https://github.com/publishpress/PublishPress-Revisions/issues/20

**Comment by JoshJourney** [comment]
This would save database space and be very helpful for communities that have thousands of pending revisions or rejected revisions. Admins would look but sometimes it can be tricky when dealing with the latest.

**Comment by stevejburge** [comment]
Thanks @Drumbell. This does sound like it's best deal with by a configurable setting

**Comment by stevejburge** [comment]
One more request for this today:

> Add a trim revision on the weekly basis option to not let the revisions accumulate.

**Comment by JoshJourney** [comment]
Weekly is rather soon especially if a team has a busy week. Would be a shame for good revisions to go so soon. Once a month could work. Perhaps an admin drop-down option Auto-Purge Duration: "Weekly, Monthly, 3 Months, Custom Days (generates temp number field)".

**Comment by stevejburge** [comment]
Thanks @DaringDreams 

Some background on those: https://wordpress.org/plugins/revisionary/advanced/. Those plugins re not very flexible with their settings. We can probably do better

**Comment by JoshJourney** [comment]
Not sure what you mean by plugins (^link)? It's a link to Revisionary which has decent setting flexibility. ;-)

**Comment by stevejburge** [comment]
Sorry, that was a bad copy-paste. Here's the intended link https://publishpress.com/blog/automatically-delete-wordpress-revisions/

**Comment by JoshJourney** [comment]
Perhaps I'm misunderstanding something but isn't a revision object a little different from a standard WP revision? According to DB table _wp_posts_ column _post_status_, each revision record from Revisions Pro has a post_status of _pending-revision_. On [Stackexchange](https://wordpress.stackexchange.com/questions/113607/how-to-remove-in-the-wordpress-database-all-posts-revisions-except-the-last-thre#351886) people commonly only query for post type 'revision'; hence potentially not affecting Revisions Pro's revisions? The plugins mentioned in your article seem as if many folks could simply use the revision [storage method](https://wordpress.org/support/article/revisions/#revision-options). However this does not address specific post types nor does it handle old posts. If the limit was 15, post's from a dozen years ago would not be auto purged if it doesn't exceed the 15 limit; hence WP's limitation behind that method.

**Comment by stevejburge** [comment]
@DaringDreams Thanks for this. I've moved the delete request to a new ticket https://github.com/publishpress/PublishPress-Revisions/issues/39

@agapetry thoughts on this topic?
